Feature/invite user modal components

b12f opened the pull request:
Still missing:

* Integration with backend services
* New/improved modal styles
* Improved button styles
* Improved selected principal "pill" styles
* Integration tests

Closes https://community.openproject.com/projects/openproject/work_packages/35521/activity

Commit:
Add initial invite user modal

Commit:
Small change to invite user modal

Commit:
Small use rmodal changes

Commit:
Builds

aleixsuau committed:
Removed meanless? .textarea-wrapper class

Commit:
Added invite user modal title translation

aleixsuau committed:
Removed useless div

Commit:
Changed modal to module

Commit:
Figured out basic modal setup

Commit:
Fixed build

Commit:
Started adding option-list component

Commit:
Adding reactive forms

Commit:
vim

Commit:
Reactive forms working for option-list

Commit:
op-form-field is implemented, project-selection step works

Commit:
Working on principal step

Commit:
Basics of principal selection are done

Commit:
Remove console logs

Commit:
Added role step and role api bindings, updated directory structure

Commit:
Added basics of message step

Commit:
Added summary page

Commit:
Add in-project check for selected principal

Commit:
Add documentation for principal filters

Commit:
Create modal-heading component

Commit:
Fix principal emit

Commit:
Add API calls to summary component

Commit:
Work on translations, rework close fn

aleixsuau committed:
Open pdf exports in new tab

oliverguenther committed:
Fix local feature testing by opt-in ENV variable in ci/docker

Commit:
Added translations

Commit:
Fixed styling

oliverguenther committed:
[35506] Avoid notifications for assigned/responsible placeholders

Right now, placeholder users cannot be watchers nor authors nor meeting
participants, so the only notifications that can happen are work package
assignees or responsibles.

https://community.openproject.com/wp/35506

aleixsuau reviewed (commented):
Hi @b12f ,

As commented in the meeting, because your proposal mixes concerns (UI behavior (Stepper) and data management (Form)) I can't approve it.

Best

ulferts committed:
Merge branch 'release/11.1' into dev

ulferts committed:
Placeholder user project members (#8961)

* remove intermediate placeholder scope

Doing so, placeholder users will begin to show up in the system

* remove scope without value

* extract scope

* use enum for status

* allow placeholder users to become project members

* display placeholder user member on members widget

* remove now superfluous method

The status name can simply be queried via #status now

* replace possible_assignees/responsibles on project

This also leads to placeholder users becoming eligible as assignees and
responsibles.

* fix aggregated scope on bulk edit

* linting

* remove IssueHelper

Commit:
Add autofocus, fix typing

Commit:
update locales from crowdin [ci skip]

tessi committed:
Fixed metadata and a typo in our GitHub Plugin (#8970)

* Fixed metadata and a typo in our GitHub Plugin

I found these while reviewing the code. Probably doesn't hurt to fix
even tiny things like these :)

https://community.openproject.com/projects/openproject/work_packages/36150

* Update modules/github_integration/lib/open_project/github_integration/engine.rb

thanks @lindenthal

oliverguenther committed:
Use SummaryLogger

Commit:
Improve typing

Commit:
Add form-binding directive to fix form-field

Commit:
Fixed typescript errors

Commit:
Fix more translations

Commit:
update locales from crowdin [ci skip]

ulferts committed:
use AR`s scope method instead of singleton class method (#8976)

* use AR`s scope method instead of singleton class method

By using the scope method, all of Rails` checks and method definition usually done on a method are now also performed on the extracted scope methods

https://github.com/rails/rails/blob/5f3ff60084ab5d5921ca3499814e4697f8350ee7/activerecord/lib/active_record/scoping/named.rb#L154-L190

* turn scope classes into modules

oliverguenther commented:
A few remarks I found while looking through it. For the sake of continuing with the integration, I suggest to merge it earlier rather than later however.

1. Long names in the ng-select will expand the width of the entire modal, such as in the second step. The same is true for the individual steps causing differences in modal width which is confusing.

![Screenshot from 2021-02-05 14-48-04](https://user-images.githubusercontent.com/459462/107041812-3d510100-67c1-11eb-961b-761419b740c5.png)

2. Styling of the buttons in the summary step is wrong
![Screenshot from 2021-02-05 14-48-22](https://user-images.githubusercontent.com/459462/107041891-5659b200-67c1-11eb-8a90-37a6bfa216bf.png)

Regarding data flow / separation: In our meeting we were still not sure if we want to treat the entire modal as one form or not. In case we do not, I'm fine with our one-way-flow approach. If we decide to have one larger form, we should switch to passing the FromGroup/FormControl to the steps instead.
